modules/light-clients/07-tendermint/types/proposal_handle.go

```diff
@@ -80,37 +59,33 @@ func (cs ClientState) CheckSubstituteAndUpdateState(
 
 	// copy consensus states and processed time from substitute to subject
 	// starting from initial height and ending on the latest height (inclusive)
-	for i := initialHeight.GetRevisionHeight(); i <= substituteClientState.GetLatestHeight().GetRevisionHeight(); i++ {
-		height := clienttypes.NewHeight(substituteClientState.GetLatestHeight().GetRevisionNumber(), i)
-
-		consensusState, err := GetConsensusState(substituteClientStore, cdc, height)
-		if err != nil {
-			// not all consensus states will be filled in
-			continue
-		}
-		SetConsensusState(subjectClientStore, cdc, consensusState, height)
+	height := substituteClientState.GetLatestHeight()
 
-		// set metadata for this consensus state
-		setConsensusMetadata(ctx, subjectClientStore, height)
```

Comment:
This was incorrect. We want to set the metadata stored for the substitute, not set it to the current context